content/master/concepts/usages.md

```diff
@@ -0,0 +1,222 @@
+to define resource instead of providing the resource name.
+
+```yaml {label="selectors"}
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: Usage
+metadata:
+  name: release-uses-cluster
+spec:
+  of:
+    apiVersion: eks.upbound.io/v1beta1
+    kind: Cluster
+    resourceSelector:
+      matchControllerRef: false # default, and could be omitted
+      matchLabels:
+        foo: bar
+  by:
+    apiVersion: helm.crossplane.io/v1beta1
+    kind: Release
+    resourceSelector:
+       matchLabels:
+          baz: qux
+```
+
+Once the `Usage` controller resolves the selectors, it persists the resource
+name in the 
+{{<hover label="selectors-resolved" line="10">}}resourceRef.name{{</hover>}}
+field. The following example shows the `Usage` resource after the resolution of
+selectors.
+
+{{<hint "important" >}}
+<!-- vale write-good.Passive = NO -->
+The selectors are resolved only once. If there are more than one matches, a
+random resource is selected from the list of matched resources.
+<!-- vale write-good.Passive = YES -->
+{{< /hint >}}
+
+```yaml {label="selectors-resolved"}
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: Usage
+metadata:
+  name: release-uses-cluster
+spec:
+  of:
+    apiVersion: eks.upbound.io/v1beta1
+    kind: Cluster
+    resourceRef:
+       name: my-cluster
+    resourceSelector:
+      matchLabels:
+        foo: bar
+  by:
+    apiVersion: helm.crossplane.io/v1beta1
+    kind: Release
+    resourceRef:
+       name: my-cluster
+    resourceSelector:
+       matchLabels:
+          baz: qux
+```
+
+## Usage in a composition
+
+A typical use case for Usages is to define a deletion ordering between the
+resources in a Composition. The Usages support
+[matching controller reference]({{<ref "./compositions#match-a-controller-reference" >}})
+in selectors to ensures that the matching resource is in the same composite
+resource in the same way as [cross-resource referencing]({{<ref "./compositions#cross-resource-references" >}}).
+
+The following example shows a Composition that defines a deletion ordering
+between a `Cluster` and a `Release` resource. The `Usage` blocks deletion of
+the `Cluster` resource until the `Release` resource is successfully deleted.
+
+```yaml {label="composition"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+spec:
+  resources:
+    - name: cluster
+      base:
+        apiVersion: container.gcp.upbound.io/v1beta1
+        kind: Cluster
+        # Removed for brevity
+    - name: release
+      base:
+        apiVersion: helm.crossplane.io/v1beta1
+        kind: Release
+        # Removed for brevity
+    - name: release-uses-cluster
+      base:
+        apiVersion: apiextensions.crossplane.io/v1alpha1
+        kind: Usage
+        spec:
+          of:
+            apiVersion: container.gcp.upbound.io/v1beta1
+            kind: Cluster
+            resourceSelector:
+              matchControllerRef: true
+          by:
```

Comment:
What happens if there are multiple Release resources in the same Composition?  Presumably we would want individual Usage resources for each and need to specify the names explicitly?

Reply:
> but not necessarily need to specify the names explicitly which may not be possible at all in a composition

Why not just patch the `metadata.name` of the resources that are being referenced into the `Usage` resource?

Reply:
Since `metadata.name` of composed resources has random suffixes (composite controller sets `metadata.generateName`), I was thinking that would require some more work like patching `metadata.name` to composite status and then patching from composite status to spec of Usage. May be I am missing something?

Reply:
No - it's definitely a 2-step process, but it's pretty common - have a ToCompositeFieldPath or ToEnvironmentFieldPath patch in the resource that puts the metadata.name in storage, and then a FromComposite or FromEnvironmentFieldPath in the Usage resource that retrieves the desired name.

Reply:
Added a tip regarding what to do when multiple of the same type.